src/server/tiering/op_manager.cc

```diff
@@ -94,13 +103,22 @@ void OpManager::ProcessStashed(EntryId id, unsigned version, DiskSegment segment
 void OpManager::ProcessRead(size_t offset, std::string_view value) {
   ReadOp* info = &pending_reads_.at(offset);
 
+  std::string key_value;
+
+  Overloaded action_resolver{
+      [&key_value](util::fb2::Future<std::string>& fut) { fut.Resolve(key_value); },
+      [&key_value](std::function<void(std::string*)>& modf) { modf(&key_value); }};
+
   for (auto& ko : info->key_ops) {
-    auto key_value = value.substr(ko.segment.offset - info->segment.offset, ko.segment.length);
+    key_value = value.substr(ko.segment.offset - info->segment.offset, ko.segment.length);
+    bool modified = false;
 
-    for (auto& fut : ko.futures)
-      fut.Resolve(std::string{key_value});
+    for (auto& action : ko.actions) {
+      std::visit(action_resolver, action);
+      modified |= std::holds_alternative<std::function<void(std::string*)>>(action);
+    }
 
-    ReportFetched(Borrowed(ko.id), key_value, ko.segment);
+    ReportFetched(Borrowed(ko.id), key_value, ko.segment, modified);
```

Comment:
add comments why `modified` is needed.

Reply:
Addded, but there are comments in ReportFetched, declaration, as well as tiered storage implementation